### ISSUE TITLE
Cleanly exit when the rasterizer can't be initialized (e.g. too low GL/GLES version)

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#15324]
**Godot version:**
Current master branch, f8b58784b.

**OS/device including version:**
Any.

**Issue description:**
Currently Godot gives an error when the rasterizer can't be initialized, and then crashes. It would be nice to have it exit cleanly after showing the error (likely once the error is acknowledged if it's a child process, to avoid killing the error dialog itself).

This could likely be done by making `RasterizerGLES3:initialize()` return `Error`, and properly abort the main init in such case (to be tested to ensure that it no longer leaks memory).

CC'ing @touilleMan who wants to work on it.

- [x] I searched the existing [GitHub issues](https://github.com/godotengine/godot/issues?utf8=%E2%9C%93&q=is%3Aissue+) for potential duplicates.

**Comment by touilleMan** [comment]
I've start working on this issue.

Here is the stacktrace an error from the rasterizer goes through:
```
* thread #1: tid = 23134, 0x00000000008aa89d godot.x11.opt.debug.64`RasterizerGLES3::initialize() + 29 at rasterizer_gles3.cpp:140, name = 'godot.x11.opt.d', stop reason = breakpoint 1.1
  * frame #0: 0x00000000008aa89d godot.x11.opt.debug.64`RasterizerGLES3::initialize() + 29 at rasterizer_gles3.cpp:140
    frame #1: 0x00000000011e9961 godot.x11.opt.debug.64`VisualServerWrapMT::init() + 49 at visual_server_wrap_mt.cpp:121
    frame #2: 0x0000000000418c74 godot.x11.opt.debug.64`OS_X11::initialize(OS::VideoMode const&, int, int) + 2276 at os_x11.cpp:506
    frame #3: 0x000000000042dd4b godot.x11.opt.debug.64`Main::setup2(unsigned long) + 75 at main.cpp:986
    frame #4: 0x000000000043213c godot.x11.opt.debug.64`Main::setup(char const*, int, char**, bool) + 11100 at main.cpp:925
    frame #5: 0x000000000040c667 godot.x11.opt.debug.64`main + 103 at godot_x11.cpp:47
    frame #6: 0x00007ffff4aa5830 libc.so.6`__libc_start_main(main=(godot.x11.opt.debug.64`main at godot_x11.cpp:38), argc=3, argv=0x00007fffffffdd88, init=<unavailable>, fini=<unavailable>, rtld_fini=<unavailable>, stack_end=0x00007fffffffdd78) + 240 at libc-start.c:291
    frame #7: 0x000000000040f539 godot.x11.opt.debug.64`_start + 41
```

So I have to modify `VisualServerWrapMT::init` to make it return an `Error`.
However in the case the visual rendering is done in it own thread, the rasterizer initialization is deferred to this side thread:

Without thread:
https://github.com/godotengine/godot/blob/92e114edfdf45aa311f7b8326d286998365b6bfa/servers/visual/visual_server_wrap_mt.cpp#L121

With thread:
https://github.com/godotengine/godot/blob/92e114edfdf45aa311f7b8326d286998365b6bfa/servers/visual/visual_server_wrap_mt.cpp#L65

So in this case we will get the segfault anyway if the init couldn't be done... So I wonder if there is a workaround to pass the information to the main thread that it should exit.

**Comment by akien-mga** [comment]
Not critical for the upcoming 3.0 release, so moving to the next milestone. A fix can still be cherry-picked for 3.0.x maintenance releases once available in the master branch.
